launch/launch/actions/replace_environment_variables.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the ReplaceEnvironmentVariables action."""
+
+from typing import Mapping
+from typing import Text
+
+from ..action import Action
+from ..launch_context import LaunchContext
+
+
+class ReplaceEnvironmentVariables(Action):
+    """
+    Action that replaces the environment variables in the current context.
+
+    The previous state can be saved by pushing the stack with the
+    :py:class:`launch.actions.PushEnvironment` action.
+    And can be restored by popping the stack with the
+    :py:class:`launch.actions.PopEnvironment` action.
+    """
+
+    def __init__(self, environment: Mapping[Text, Text], **kwargs) -> None:
+        """Create a ReplaceEnvironmentVariables action."""
+        super().__init__(**kwargs)
+        self.__environment = environment
+
```

Comment:
Lets add the @classmethod parse () method here, so that this works in xml/yaml style parsing as well. Also a test for that case. For reference : https://github.com/ros2/launch/blob/dd6ab0edf19c2f35b8e71aa8151df50b8a2feff9/launch/launch/actions/reset_environment.py#L39

Reply:
I think that `ReplaceEnvironmentVariables` should not be exposed to xml and yaml frontends. It's similar to `PopEnvironment` and `PushEnvironment` in that it should only be used by Launch internally and not by its users. But maybe it's just me who can't find a reason to expose it, so I made the suggested changes.

Please let me know if anything else is needed.

Reply:
Yeah, now that I think that makes sense.